lib/views/explore/explore_page.dart

```diff
@@ -0,0 +1,184 @@
+  late Spaces _spaces;
+
+  @override
+  void initState() {
+    super.initState();
+    _spaces = Spaces.newest;
+  }
+
+  String getStateSpace() {
+    switch (_spaces) {
+      case Spaces.newest:
+        return "Newest";
+      case Spaces.active:
+        return "Recently Active";
+      case Spaces.mostParticipants:
+        return "Most Participants";
+      case Spaces.mostMessages:
+        return "Most Messages";
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return PageSkeleton(
+      appBarChildren: [
+        const BB.BackButton(),
+        const SizedBox(width: 24),
+        Text(
+          "Explore Spaces",
+          style: GoogleFonts.montserrat(
+            fontSize: 20,
+            fontWeight: FontWeight.w800,
+          ),
+        ),
+        const Spacer(),
+      ],
+      bodyChildren: [
+        const SizedBox(height: 8),
+        Row(
+          children: [
+            GestureDetector(
+              onTap: () {
+                showModalBottomSheet(
+                  context: context,
+                  builder: (context) {
+                    return SizedBox(
+                      height: getHeight(context) * 0.3,
+                      child: Column(
+                        children: [
+                          ListTile(
+                            title: const Text("Newest"),
+                            onTap: () {
+                              setState(() {
+                                _spaces = Spaces.newest;
+                              });
+                              Navigator.pop(context);
+                            },
+                          ),
+                          ListTile(
+                            title: const Text("Recently Active"),
+                            onTap: () {
+                              setState(() {
+                                _spaces = Spaces.active;
+                              });
+                              Navigator.pop(context);
+                            },
+                          ),
+                          ListTile(
+                            title: const Text("Most Participants"),
+                            onTap: () {
+                              setState(() {
+                                _spaces = Spaces.mostParticipants;
+                              });
+                              Navigator.pop(context);
+                            },
+                          ),
+                          ListTile(
+                            title: const Text("Most Messages"),
+                            onTap: () {
+                              setState(() {
+                                _spaces = Spaces.mostMessages;
+                              });
+                              Navigator.pop(context);
+                            },
+                          ),
+                        ],
+                      ),
+                    );
+                  },
+                );
```

Comment:
it will be better to separate this modalsheet code to a different function

Reply:
Good point, done!